Odata-docs/webapi-8/tutorials/custom-skiptokenhandler.md

```diff
@@ -0,0 +1,549 @@
+The logic for building the Edm model and configuring the OData service is as follows:
+
+# [.NET 6.0](#tab/net60)
+
+```csharp
+// Program.cs
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.OData;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.OData.ModelBuilder;
+using CustomSkipTokenSample.Models;
+
+var builder = WebApplication.CreateBuilder(args);
+
+var modelBuilder = new ODataConventionModelBuilder();
+modelBuilder.EntitySet<Customer>("Customers");
+
+builder.Services.AddControllers().AddOData(
+    options => options.SkipToken().AddRouteComponents(
+        "odata",
+        modelBuilder.GetEdmModel()));
+
+var app = builder.Build();
+
+app.UseRouting();
+app.UseEndpoints(endpoints => endpoints.MapControllers());
+
+app.Run();
+```
+
+# [.NET Core 3.1](#tab/netcoreapp31)
+
+```csharp
+// Startup.cs
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.OData;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.OData.ModelBuilder;
+using RefRouting.Models;
+
+public class Startup
+{
+    public void ConfigureServices(IServiceCollection services)
+    {
+        var modelBuilder = new ODataConventionModelBuilder();
+        modelBuilder.EntitySet<Customer>("Customers");
+
+        services.AddControllers().AddOData(
+            options => options.SkipToken().AddRouteComponents(
+                "odata",
+                modelBuilder.GetEdmModel()));
+    }
+
+    public void Configure(IApplicationBuilder app)
+    {
+        app.UseRouting();
+        app.UseEndpoints(endpoints => endpoints.MapControllers());
+    }
+}
+```
+
+---
+
+The important thing to note here is that we enable the `$skiptoken` query option by calling the `options.SkipToken()` method (we can also enable all query features using `options.EnableQueryFeatures(null)`).
+
+## Controllers
+
+We create a simple controller for the `Customers` entity set. We include some dummy data in the controller for simplicity:
+
+```c#
+public class CustomersController : ODataController
+{
+    private static List<Customer> customers = new List<Customer>
+    {
+        new Customer
+        {
+            Id = 1,
+            Name = "Customer 1"
+        },
+        new Customer
+        {
+            Id = 2,
+            Name = "Customer 2"
+        },
+        new Customer
+        {
+            Id = 3,
+            Name = "Customer 3"
+        },
+        new Customer
+        {
+            Id = 4,
+            Name = "Customer 4"
+        },
+        new Customer
+        {
+            Id = 5,
+            Name = "Customer 5"
+        }
+    };
+
+    [EnableQuery(PageSize = 2)]
+    public ActionResult Get()
+    {
+        return customers;
+    }
+}
+```
+
+The important thing to note here is that applied `[EnableQuery]` attribute to our controller's `Get` method and set the `PageSize` property to `2`. This means that the server will send up to two items per response page for requests made to `GET /odata/Customers`.
```

Comment:
"is that applied `[EnableQuery]` attribute to" - the grammar is not quite right...

Reply:
Rephrased to "is that we applied..."